### PR TITLE
test(combineLatest): add tests with empty or never sources

### PR DESCRIPTION
**Pull request by adrianomelo**
Observable.combileLatest should be never when:
 - sources are never and never
 - one of the sources is never
 - one of the sources is never and the other has values
 - one of the sources is never and the other complete
 - sources are empty and empty
 - one of the sources is empty
 - one of the sources is empty and the other complete

with two failing tests:
 - Observable.combineLatest should be never when sources are empty and empty
 - Observable.combineLatest should be never when one of the sources is empty and the other complete

### CODE REVIEW

**Comment by adrianomelo** [comment]
Hi guys,

This is my first patch with code to this project..
There are two failing tests in this patch, I would like to know your opinion about which code is incorrect.. combineLatest or these 2 tests?

File: spec/observables/combineLatest-spec.js [hunk 1]
```diff
@@ -3,15 +3,69 @@ var Rx = require('../../dist/cjs/Rx');
 var Observable = Rx.Observable;
 
 describe('Observable.combineLatest', function () {
+  function concat(x, y) { return '' + x + y; }
+
+  it('should be never when sources are never and never', function () {
```
Comment (kwonoj) [inline spec/observables/combineLatest-spec.js]: Isn't this case already exists in [spec/operators/combineLatest-spec](https://github.com/ReactiveX/RxJS/blob/master/spec/operators/combineLatest-spec.js#L77)?

Comment (adrianomelo) [inline spec/observables/combineLatest-spec.js]: You are right! I didn't know this file.. I'll close this PR and take a look in this file.
